### ISSUE TITLE
I cant update my strapi from 4.14.5 to 4.20.1

### ISSUE THREAD

**Issue by ingmanuelcaicedo** [strapi/strapi#19564]
### Required System information

- Node.js version: 18.18.2
- NPM version: 9.6.4
- Strapi version: 4.14.5
- Database: SQLite
- Operating system: Windows 11
- Is your project Javascript or Typescript: Javascript

### Steps to reproduce the behavior

I currently have a Strapi functional installation with v4.14.15, I would like to update to the last version, and I follow the next steps.

1. Update strapi to 4.20.1 on packackge.json the next dependencies: 
    - @strapi/plugin-graphql
    - @strapi/plugin-i18n
    - @strapi/plugin-users-permissions
    - @strapi/strapi
3. Save the file.
4. Go to terminal and run ```npm run install```
5. Installing dependencies shows the next message:
    -  ![Screenshot_1](https://github.com/strapi/strapi/assets/12139208/59872c45-d295-475d-ba6e-fbcdabdedd7c)
    - I skip these, because they aren't requiered to continue.
7. Run ```npm run build``` and then, terminal asks about installing some packages, I typed **Yes**:
    - The command starts installing before building:
    - ![Screenshot_3](https://github.com/strapi/strapi/assets/12139208/3dbfe7b6-5e06-4c65-b5c1-614b084826f6)
    - After a pair of seconds the terminal sends those error messages:
    - ![Screenshot_4](https://github.com/strapi/strapi/assets/12139208/edd2bd43-9b98-4e74-9b59-ead6912f6247)
8. After that, the project can't be build and the package.json file has the new requiered packages now.
    - ![Screenshot_5](https://github.com/strapi/strapi/assets/12139208/e480ebb0-1544-4383-b613-6d4fa557f6b5)
10. s

### Expected behavior
I expected to get my project built but it didn't happened.

### Screenshots
- Step 5: ![Screenshot_1](https://github.com/strapi/strapi/assets/12139208/59872c45-d295-475d-ba6e-fbcdabdedd7c)
- Step 7: ![Screenshot_3](https://github.com/strapi/strapi/assets/12139208/3dbfe7b6-5e06-4c65-b5c1-614b084826f6)
- Step 7: ![Screenshot_4](https://github.com/strapi/strapi/assets/12139208/edd2bd43-9b98-4e74-9b59-ead6912f6247)
- Step 8: ![Screenshot_5](https://github.com/strapi/strapi/assets/12139208/e480ebb0-1544-4383-b613-6d4fa557f6b5)

### Code snippets
First I ran: 
``` npm install ```

Then I ran:
``` npm run build ```

And got: 
```
> example--strapi@0.5.000 build
> strapi build

[WARN] [@strapi/strapi]: The NODE_ENV is not set to production. This may result in unexpected behavior.
⠋ Building build context
⠙ Building build context[INFO] Including the following ENV variables as part of the JS bundle:
    - ADMIN_PATH
    - STRAPI_ADMIN_BACKEND_URL
    - STRAPI_TELEMETRY_DISABLED
✔ Building build context (195ms)
⠋ Building admin panel
⠧ Building admin panel[ERROR] Module not found: Error: Can't resolve '@strapi/plugin-graphql/strapi-admin' in 'D:\Development\strapi\example\.strapi\client'
✖ Building admin panel
[ERROR]  There seems to be an unexpected error, try again with --debug for more information
```

### Additional context
I tried to make a new installation of strapi as an example project, running the next command:

``` npx create-strapi-app@latest example --quickstart ```

And also ran the next command:

``` npm run build ```

But I got the same error as you can see in the next screenshot:

![Screenshot_6](https://github.com/strapi/strapi/assets/12139208/8803189e-d21b-4f06-b5a3-384828d477eb)

Does anyone has the same issue?

**Comment by derrickmehaffy** [comment]
One your plugins is trying to import a react component we don't export and never intended to.

I'd recommend starting by removing some plugins one by one and see which one is doing it.

**Comment by ingmanuelcaicedo** [comment]
> One your plugins is trying to import a react component we don't export and never intended to.
> 
> I'd recommend starting by removing some plugins one by one and see which one is doing it.

Thank you so much for the answer, but it's something strange because I have the same problem installing a new clean installation. On the other hand, I was reading in another issue published here, that the problem is plugin-i18n but I need to have this plugin, because I use translation in my Strapi app.

**Comment by ingmanuelcaicedo** [comment]
Hello @derrickmehaffy AS I was saying, I have this problem with a new installation also.
For this example, I just have uploaded a new installation of Strapi on [This Repo](https://github.com/ingmanuelcaicedo/example-strapi-4-20-1) and I also got the same error. What could happend in this case???

**Comment by jkefc** [comment]
I agree. In my case it was the plugin "@notum-cz/strapi-plugin-record-locking" - a few months ago when I was starting the project, the most recent version was "0.3.4", but since then they managed to do a few fixes, decided to do a major release, and few more fixes, so as soon as I upgraded to version 1.3.10, everything started to work fine.

**Comment by derrickmehaffy** [comment]
Going to go ahead and close this as it appears resolved
